lib/mosaic-geomath/src/main/java/org/eclipse/mosaic/lib/spatial/Grid.java

```diff
@@ -130,22 +131,22 @@ public boolean addItem(T item) {
 
     public void updateGrid() {
         synchronized (tmpIndexA) {
-            items.forEach((item, currentIndex) -> {
-                CellIndex newCellIndex = toCellIndex(adapter.getCenterX(item), adapter.getCenterZ(item), tmpIndexA);
+            items.forEach((key, currentIndex) -> {
+                CellIndex newCellIndex = toCellIndex(adapter.getCenterX(key.item), adapter.getCenterZ(key.item), tmpIndexA);
                 if (newCellIndex.isEqualTo(currentIndex)) {
                     // no index change -> do nothing
                     return;
                 }
-                getGridCell(currentIndex).remove(item);
+                getGridCell(currentIndex).remove(key.item);
                 currentIndex.set(newCellIndex);
-                getGridCell(currentIndex).add(item);
+                getGridCell(currentIndex).add(key.item);
             });
         }
     }
 
     public void removeItem(T item) {
         synchronized (tmpIndexA) {
-            CellIndex cellIndex = items.remove(item);
+            CellIndex cellIndex = items.remove(new ItemKey<>(adapter.getItemHash(item), item));
```

Comment:
Isn't this a little bit dirty? We have to instantiate a new `ItemKey` whenever we want to remove an item. 🤷 Also this is a duplicate of line 122. Maybe we can somewhat hide it by adding a method with the structure `ItemKey<T> getKeyForItem(T Item)`

Reply:
I didn't like either but I could not think of a better way. Performance/memory wise it should be okay to create this object just for removing the item. I followed your suggestions regarding hiding the instantiation.